File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {

        map<int, int> hash_map;

        for(int i = 0; i < nums.size(); i++){
            hash_map[nums.at(i)]++;
        }

        for(auto &x : hash_map) {
            if (x.second > 1) {
                return true;
            } 
        }
        return false;
        

    }
};

File: Data Structures & Algorithms/is-anagram/submission-4.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False
        
        hash_map = {}

        for i in range(len(s)):
            hash_map[s[i]] = hash_map.get(s[i], 0) + 1
            hash_map[t[i]] = hash_map.get(t[i], 0) - 1
        
        return not any(hash_map.values())

        